### PR TITLE
Write QGC-compatible param files

### PR DESCRIPTION
**Pull request by clydemcqueen**
Make it easy to generate QGC-compatible param files from tlog files.

Also fixed a bug in expand_path().

Next PR is to add some tests.

### CODE REVIEW

**Comment by clydemcqueen** [comment]
Hey @billmania -- review, or not, at will. I think the next PR will focus on tests and github CI.

File: tlog_param.py [hunk 1]
```diff
@@ -1,39 +1,97 @@
...
+                self.autopilot_version = f'{major}.{minor}.{path} {firmware_version_type_str(version_type)}'
+                self.git_hash = bytes(data['flight_custom_version']).decode('utf-8')
+
+    def write(self, outfile: str):
+        """
+        Write a QGC-compatible params file
+
+        File format: https://dev.qgroundcontrol.com/master/en/file_formats/parameters.html
+        """
+        if len(self.params) == 0:
```
Comment (billmania) [inline tlog_param.py:74]:
```
if not self.params:
```

is a bit more Pythonic, assuming None, '', and {} all mean there's nothing to print.

File: tlog_param.py [hunk 2]
```diff
@@ -1,39 +1,97 @@
...
+    if firmware_version_type == mav_common.FIRMWARE_VERSION_TYPE_DEV:
+        return 'dev'
+    elif firmware_version_type == mav_common.FIRMWARE_VERSION_TYPE_ALPHA:
+        return 'alpha'
+    elif firmware_version_type == mav_common.FIRMWARE_VERSION_TYPE_BETA:
+        return 'beta'
+    elif firmware_version_type == mav_common.FIRMWARE_VERSION_TYPE_RC:
+        return 'rc'
+    else:
+        return ''
```
[the remark this answers is not in the document]
Comment (clydemcqueen) [reply tlog_param.py:31]: Nice! Thanks

[the remark this answers is not in the document]
Comment (clydemcqueen) [reply tlog_param.py]: I went w/ the description field, as I think it's a bit nicer for humans. E.g., 'official stable release' vs 'FIRMWARE_VERSION_TYPE_OFFICIAL'.

File: tlog_param.py (hunk 1 above)
Comment (clydemcqueen) [reply tlog_param.py]: Done

File: tlog_param.py [hunk 3]
```diff
@@ -1,39 +1,97 @@
...
+        f.write('# Onboard parameters for Vehicle 1\n')
+        f.write('#\n')
+        f.write('# Stack: ArduPilot\n')
+        f.write('# Vehicle: Sub\n')
+        f.write(f'# Version: {self.autopilot_version}\n')
+        f.write(f'# Git Revision: {self.git_hash}\n')
+        f.write('#\n')
+        f.write('# Vehicle-Id\tComponent-Id\tName\tValue\tType\n')
+
+        for pi in sorted(self.params.items()):
```
[the remark this answers is not in the document]
Comment (clydemcqueen) [reply tlog_param.py]: Done

**Pull request merged**